src/Chapter3.hs

```diff
@@ -480,6 +510,35 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+type CastleName = String
+data Wall = Wall deriving (Show)
+data Castle = Castle CastleName (Maybe Wall) deriving (Show)
+data Sight = Church | Library deriving (Show)
+data NumberOfPeople = One | Two | Three | Four deriving (Show, Enum)
+data House = House NumberOfPeople deriving (Show)
+data City = City { cityCastle :: Maybe Castle
+                 , citySight :: Sight
+                 , cityHouses :: [House]
+                 } deriving (Show)
```

Comment:
This is a nice design 🏗️ 👏🏼 
It is a very nice combination of using standard types and creating your own.

Sometimes, it could make life a bit easier to create your own times. For example, in here I see the opportunity to ease the work with `Maybe Castle`.
As you correctly noticed, there are three states of the castle:

1. Castle with a name
2. Castle with the name + walls
3. No castle

This actually could be written in the sum type:

```haskell
data Castle 
    = Castle String
    | WallCastle String
    | None
```

With such representation when you don't have nested `Maybe` some of the functions could be simplified 🙂

Reply:
I thought about it, I even started with something like this, but then for some reason I decided to use Maybe